### PR TITLE
added Self, Height and Weight classes

### LINKED ISSUE #117: Create Self Class with Attributes for Height and Weight

### PR DESCRIPTION
**Pull request by Jarrod-Bob**

### CODE REVIEW

File: src/main/java/seedu/address/model/self/Weight.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents the user's weight in the diet tracker of EYLAH.
+ * Guarantees: immutable; is valid as declared in {@link #isValidWeight(String)}
+ */
+public class Weight {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Weight should only contain numbers, and it should not be blank. It is to be input in kilograms.";
```
Comment (WillySeahh) [inline src/main/java/seedu/address/model/self/Weight.java:13]: hmm maybe it should be better to phrase it like ".... The input should be in kilograms."

File: src/main/java/seedu/address/model/self/Self.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Self {
+
+    // Data fields
+    private final Height height;
+    private final Weight weight;
+
+    /**
+     * Every field must be present
```
Comment (WillySeahh) [inline src/main/java/seedu/address/model/self/Self.java:18]: I think its good to add the @param in the javadocs for this method.

File: src/main/java/seedu/address/model/self/Height.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+    /**
+     * Constructs a {@code Height}.
+     *
+     * @param height A valid height.
+     */
+    public Height(String height) {
+        requireNonNull(height);
+        checkArgument(isValidHeight(height), MESSAGE_CONSTRAINTS);
+        this.height = height.toString();
```
Comment (akhilvuputuri) [inline src/main/java/seedu/address/model/self/Height.java]: No need to call toString on the height as it is already a String. Causing Codacy errors.

File: src/main/java/seedu/address/model/self/Weight.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+    /**
+     * Constructs a {@code Weight}.
+     *
+     * @param weight A valid weight.
+     */
+    public Weight(String weight) {
+        requireNonNull(weight);
+        checkArgument(isValidWeight(weight), MESSAGE_CONSTRAINTS);
+        this.weight = weight.toString();
```
Comment (akhilvuputuri) [inline src/main/java/seedu/address/model/self/Weight.java]: Same as height, no need to call toString on the weight as it is already a String. Causing Codacy errors.

**Review by sheexiong — APPROVED** [review]
LGTM

**Review by akhilvuputuri — APPROVED** [review]

**Pull request merged**